### ISSUE TITLE
Add support for Gradle 7.4 Version Catalogs

### ISSUE THREAD

**Issue by DanySK** [jmfayard/refreshVersions#333]
Gradle 7.0 introduced the "[centralized declaration of dependencies](https://docs.gradle.org/7.0/release-notes.html#centralized-dependency-versions)".
Namely, dependencies can be aliased, grouped, and versioned in a TOML file.
It would be incredibly nice if refreshVersions could target the `libs.versions.toml` file in future, rather than relying on a custom `version.properties`.

## Update: notes from jmfayard

the plan for now is not to give up `versions.properties`

it's to auto-generate the libraries versions catalog, with the placeholder version `_` and use the typesafe accessors to help migrate an existing project to refreshVersions

Running `$ ./gradlew refreshVersionsCatalog` would

   - Enable catalog version if necessary

```
// Enable Gradle's version catalog support
// https://docs.gradle.org/current/userguide/platforms.html
enableFeaturePreview("VERSION_CATALOGS")
```

   - Generate versions.properties, with all version keys (marked as unused)
   - Generate `gradle/libs.dependencies.toml`
   - Use the IDE and auto-completion to replace hardcoded dependencies with refreshVerion's dependency notations and versionCatalog or buildSrc/Libs.kt
   - To help with this, this task will show which typesafe accessors are available `$ ./ gradlew refreshVersionsMapping`

```
androidx.car:car                             | AndroidX.car
com.retrofit:retrofit-converter-json          | Square.retrofit.converter.json
org.koin:koin                                  |  libs.koin
```

Documentation

- [Sharing dependency versions between projects](https://docs.gradle.org/7.0-rc-1/userguide/platforms.html#sub:central-declaration-of-dependencies)
- [Cédric Champeau's blog: Simplified version management with Gradle 7](https://melix.github.io/blog/2021/03/version-catalogs.html)
- [tivi/libs.versions.toml at 4e378d36d37a878060bc824ab63adcc5408b203b · chrisbanes/tivi](https://github.com/chrisbanes/tivi/blob/4e378d36d37a878060bc824ab63adcc5408b203b/gradle/libs.versions.toml)

**Comment by LouisCAD** [comment]
Hello, this is already in the mental backlog. A few things in the incubating versions catalog feature makes it that it'd not allow to have an ideal experience with refreshVersions, but it's still planned to experiment with it and most likely send feedback to Gradle on it, so it keeps incubating towards "a grown up feature".

**Comment by jmfayard** [comment]
One use-case here
https://github.com/chrisbanes/tivi/blob/main/gradle/libs.versions.toml

**Comment by jmfayard** [comment]
complete documentation is available here
https://docs.gradle.org/7.0/userguide/platforms.html#sub:central-declaration-of-dependencies

**Comment by jmfayard** [comment]
I have a plan to combine versions catalog and refreshVersions. I updated the issue with my notes

**Comment by LouisCAD** [comment]
Gradle 7.4, freshly released, finally brings improvements and stabilization for version catalogs, which is helping us move this issue forward.

First step is coming in #504: that one will still work with the version placeholder.

However, we understand that the initial request is about supporting plain version catalogs, that contain the versions. It is technically possible since TOML supports comments, so we'll consider it after #504 is behind us.

**Comment by jmfayard** [comment]
I've got news! cc @DanySK  @beski @jsschneider @NikkyAI and all.

I implemented support for versions catalog:
- generating it automatically
- rewriting the files `build.gradle(.kts)` to use them
- adding dependency updates in `gradle/libs.versions.toml`

It's published in the snapshot version `"0.40.2-SNAPSHOT"`

Have a look at the **Testing?** pargraph from the pull-request https://github.com/jmfayard/refreshVersions/pull/509
